### PR TITLE
implemented user login api | add JWT token generation

### PR DESCRIPTION
**Pull request by VasylAzarov**
implemented user login api | add JWT token generation

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -144,5 +144,22 @@
...
+        </dependency>
+        <dependency>
+            <groupId>io.jsonwebtoken</groupId>
+            <artifactId>jjwt-impl</artifactId>
+            <version>0.12.6</version>
+            <scope>runtime</scope>
+        </dependency>
+        <dependency>
+            <groupId>io.jsonwebtoken</groupId>
+            <artifactId>jjwt-jackson</artifactId> <!-- Для работы с JSON -->
```
Comment (liliia-ponomarenko) [inline pom.xml]:
```suggestion
            <artifactId>jjwt-jackson</artifactId>
```

File: src/main/java/dev/vasyl/proj/security/JwtAuthenticationFilter.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+                    userDetails, null, userDetails.getAuthorities()
+            );
+            SecurityContextHolder.getContext().setAuthentication(authentication);
+        }
+
+        filterChain.doFilter(request, response);
+    }
+
+    private String getToken(HttpServletRequest request) {
+        String bearerToken = request.getHeader("Authorization");
```
Comment (liliia-ponomarenko) [inline src/main/java/dev/vasyl/proj/security/JwtAuthenticationFilter.java]: HttpHeader.AUTHORIZATION

File: src/main/java/dev/vasyl/proj/security/JwtAuthenticationFilter.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+        if (StringUtils.hasText(bearerToken) && bearerToken.startsWith("Bearer ")) {
```
Comment (liliia-ponomarenko) [inline src/main/java/dev/vasyl/proj/security/JwtAuthenticationFilter.java]: make this variable constant and you can use length() method

File: src/main/java/dev/vasyl/proj/dto/user/UserLoginResponseDto.java [hunk 4]
```diff
@@ -0,0 +1,16 @@
+package dev.vasyl.proj.dto.user;
+
+import io.swagger.v3.oas.annotations.media.Schema;
+import lombok.AllArgsConstructor;
+import lombok.Data;
+
+@Data
+@AllArgsConstructor
+public class UserLoginResponseDto {
```
Comment (liliia-ponomarenko) [inline src/main/java/dev/vasyl/proj/dto/user/UserLoginResponseDto.java]: Can it be a record class?

**Review by liliia-ponomarenko — CHANGES_REQUESTED** [review]
Good job! Let’s fix a few comments ;)

**Review by sarakhmen — APPROVED** [review]
lgtm!

**Pull request merged**